gobblin-runtime/src/main/java/org/apache/gobblin/util/MySQLStoreUtils.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.util;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+
+
+/**
+ * MySQL based implementations of stores require common functionality that can be stored in a utility class. The
+ * functionality includes executing prepared statements on a data source object and executing SQL queries at fixed
+ * intervals. The instantiater of the class should provide the data source used within this utility.
+ */
+public class MySQLStoreUtils {
```

Comment:
naming-wise, what's mysql-specific about the impl?  wouldn't it work for any DB?  (is it merely for that fancy `HikariDataSource` logging trick)  if that's all, let's not codify mysql in the name...

Reply:
DBStatementExecutor has a nice ring to it